### PR TITLE
Make sqltools work in Theia. ConfigManager.completionLanguages is undefined

### PR DESCRIPTION
**Pull request by ariel-bentu**
This fix, is required to run properly in eclipse/theia. It is a right (IMHO) fix by itself.
-> fix exception caused when ConfigManager.completionLanguages is undefined.

### CODE REVIEW

File: packages/extension/language-client/client.ts [hunk 1]
```diff
@@ -101,9 +101,16 @@ export class SQLToolsLanguageClient implements SQLTools.LanguageClientInterface
         version: VSCodeVersion,
       },
     };
-    const selector = ConfigManager.completionLanguages
-      .concat(ConfigManager.formatLanguages)
-      .reduce((agg, language) => {
+    let selector = [];
+    if (ConfigManager.completionLanguages){
+      selector.concat(ConfigManager.completionLanguages);
```
Comment (mtxr) [inline packages/extension/language-client/client.ts]: `array.concat` doesnt change the original array. So we need to assign it back to `selector` or use `push`.

`selector = selector.concat(ConfigManager.completionLanguages);`
or 
`selector.push(...ConfigManager.completionLanguages);`

File: packages/extension/language-client/client.ts [hunk 2]
```diff
@@ -101,9 +101,16 @@ export class SQLToolsLanguageClient implements SQLTools.LanguageClientInterface
...
+    }
+
+    if (ConfigManager.formatLanguages) {
+      selector.concat(ConfigManager.formatLanguages);
```
Comment (mtxr) [inline packages/extension/language-client/client.ts]: same here

**Comment by mtxr** [comment]
Once you do those fixes, I will merge it.

I have some changes to do till tomorrow, I plan to release it tomorrow/Monday.

SAP Hana will be live on this release! :tada:

File: packages/extension/language-client/client.ts (hunk 1 above)
Comment (ariel-bentu) [reply packages/extension/language-client/client.ts]: sure. stupid me...
was so happy it works, and no way to check the specific logic.

File: packages/extension/language-client/client.ts (hunk 2 above)
Comment (ariel-bentu) [reply packages/extension/language-client/client.ts]: same

**Comment by ariel-bentu** [comment]
I still see some things that do not work in Theia. I will probably keep working on it tomorrow. If I find issue at theia - I contribute to them. like this: https://github.com/theia-ide/theia/pull/5185. and if I see I can make sqltools better and also work in theia I do it here :)
Thanks

**Comment by mtxr** [comment]
Pretty cool!! I wasn't aware of that project before, but it's really interesting!

Keep going, just let me know once you think it's available for merging, I'll be reviewing as you code.

Thanks @ariel-bentu !

**Comment by mtxr** [comment]
Well, I'll merge this small part for now since it already fixes some code

**Pull request merged**